Server auth

bcolucci opened the pull request:
** BASED ON THE BRANCH "removing-gin" **
** MERGE FIRST "removing-gin" **

- Add a HTTP Basic Auth (digest) mecanism to secure the BeePing server.
- Update the readme file.

TODO: Update the dependencies lock (don't know how to do, sorry...)

bcolucci committed:
removing gin

bcolucci committed:
remove usless error check when unmarshall basic struct Beeping

bcolucci committed:
Merge pull request #1 from yanc0/master

Merge locally

bcolucci committed:
merging

bcolucci committed:
- Add a http basic auth (digest) mechanism to secure the server
- Update the readme

bcolucci committed:
fix bug when checking the auth option value

yanc0 commented:
`dep ensure` to update your Gopkg

bcolucci committed:
update deps lock

bcolucci commented:
Dependencies lock updated.

yanc0 commented:
Thanks Brice! 

Here some examples for testing this PR.

![digest-test](https://user-images.githubusercontent.com/3806094/28017156-18fd0848-6578-11e7-91bd-eeefa27552d2.png) 
![digest2](https://user-images.githubusercontent.com/3806094/28018433-98fc5ea4-657d-11e7-8603-074b873b0edc.png)


```
$ ./beeping -listen "[2002:c000:203::1]" -auth "admin:0821726485a84fcb0d4c90fecf5d40b6"
$ curl -6 '[2002:c000:203::1]:8080/check' -d '{"url": "http://www.nyan.cat/"}' -u "admin:passw0rd" --digest
```

But digest is a bit complicated if you don't really know how to set correctly Realm or DigestDomain.

This is why I generally implement BasicAuth because it's easier to setup and with a strong password + HTTPS there is no security problem (maybe @insp3ctre can argue on this point).

Also, there is no need for external dependencies, look at this function:
https://github.com/yanc0/greedee/blob/master/http.go#L16-L32

Digest is OK for me but I think we have to document (README) how to activate BeePing Auth more precisely with command line only (MacOS and GNU/Linux)?

What do you think guys?

Cheers ! 
Yann

bcolucci commented:
I thought we can add the -auth-method argument. By default it could be "basic", and we could have "digest".

yanc0 commented:
Seems OK for me.

bcolucci committed:
- Split arguments for server auth mechanism
- The default digest method is now 'clear' (no hashing)
- Remove the go-http-auth dependency

bcolucci commented:
Just made some changes. Please do not merge until I add at least md5 digest mechanism :)
